### PR TITLE
Add translation information to database

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#19047|

## Description

This PR adds the required changes to create and modify translations entries in a rocksdb database.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Source upgrade
- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+public:
+    /**
+     * @brief Creates or modifies a translation entry in a rocksdb database.
+     *
+     * @param data JSON with the required information to create or update a translation entry.
+     * @param rocksDbWrapper rocksdb wrapper instance.
+     */
+    static void upsertTranslationEntry(const nlohmann::json& data, Utils::RocksDBWrapper& rocksDbWrapper)
+    {
+        if (data.contains("data"))
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:138]: If "data" variable doesn't contain "data" field, should be printed an invalid data format error message to std::cerr (like in each item)?

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        {
+            std::cerr << "Key already exists." << std::endl;
+        }
+        else if (data.contains("payload"))
+        {
+            std::string flatbufferSchemaStr;
+            bool valid = (flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr));
+            if (!valid)
+            {
+                std::cerr << "Couldn't open flatbuffer schema: " << FLATBUFFER_PATH << std::endl;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:53]: The method should return here if "flatbufferSchemaStr" variable is not valid

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            }
+
+            const std::string payload = data.at("payload").dump();
+            const std::string resource = data.at("resource").get<std::string>();
+            flatbuffers::Parser parser;
+
+            valid = (parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES) && parser.Parse(payload.c_str()));
+            if (!valid)
+            {
+                std::cerr << "Error: " << parser.error_ << std::endl;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:63]: Shouldn't this method return here if parsing was not valid?

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+     * @param rocksDbWrapper rocksdb wrapper instance.
+     */
+    static void updateTranslationEntry(const nlohmann::json& data, Utils::RocksDBWrapper& rocksDbWrapper)
+    {
+        std::string translationEntryStr;
+        if (rocksDbWrapper.get(data.at("resource").get<std::string>(), translationEntryStr))
+        {
+            const std::string resource = data.at("resource").get<std::string>();
+            std::string flatbufferSchemaStr;
+            flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr);
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:91]: Why the flatbuffers::LoadFile return value is not checked for validity here and in the insert method it was?

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            flatbuffers::Parser parser;
+            parser.opts.strict_json = true;
+
+            bool valid = parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES);
+            if (!valid)
+            {
+                std::cerr << parser.error_ << std::endl;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:98]: Shouldn't this method return here if parsing was not valid?

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            try
+            {
+                nlohmann::json translationEntryJSON = nlohmann::json::parse(translationEntry);
+                nlohmann::json translationEntryJSONPatched = translationEntryJSON.patch(data.at("operations"));
+                std::string translationEntryJSONPatchedStr = translationEntryJSONPatched.dump();
+                valid = parser.Parse(translationEntryJSONPatchedStr.c_str());
+
+                if (!valid)
+                {
+                    std::cerr << parser.error_ << std::endl;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:113]: Shouldn't this method return here if parsing was not valid?

File: src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs:1]: This file shouldn't be the same as the one from branch issue #19044 of @GabrielEValenzuela?

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs:1]: Yes, we need to synchronize and use this format. Possible with an array of actions.

https://github.com/wazuh/wazuh/blob/9ecdf7270c3be0725d03fee1ccec20f4e995af2a/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityTranslationPackages.fbs#L1-L23

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 2 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp]: Changed

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 3 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp]: Yeap, changed

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 4 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp]: Mistake, changed.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 5 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp]: Fixed

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 6 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:113]: Changed

File: src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs:1]: I suggested using ENUMs for this, but I was told to use std::string.

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs:1]: Do you mean for the `action` field?

Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/schemas/translation.fbs:1]: Yeap

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+                else
+                {
+                    std::string translationEntry;
+                    if (!flatbuffers::GenText(parser, translationEntryStr.c_str(), &translationEntry))
+                    {
+                        try
+                        {
+                            nlohmann::json translationEntryJSON = nlohmann::json::parse(translationEntry);
+                            nlohmann::json translationEntryJSONPatched =
+                                translationEntryJSON.patch(data.at("operations"));
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:124]: Using `https://json.nlohmann.me/api/basic_json/patch_inplace/` we can avoid a copy.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+        bool ret = false;
+
+        if (rocksDbWrapper.get(data.at("resource").get<std::string>(), translationEntryStr))
+        {
+            std::cerr << "Key already exists." << std::endl;
+        }
+        else if (data.contains("payload"))
+        {
+            std::string flatbufferSchemaStr;
+            bool valid = (flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr));
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:52]: You're right, I thought about that while doing the test but I didn't go so much further regarding the binary once the feature is released. I think that at this moment there are a lot of features that only rely on UTs and are not fully functional in an e2e. 
To not delay even more this PR I suggest addressing this later with many other improvements. 

@Dwordcito

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 7 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:124]: I get this error, i need to review the implementation 
image

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:52]: I agree, we'll fix this in another PR.
Thank you.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp (hunk 7 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp:124]: I don't know why it isn't working.
We can ignore the suggestion.

**Pull request merged**